File: GrpProj1/Blackjack code in C++/Blackjack-master/Player.cpp
/* 
 * File:   Player.cpp
 * Author: Kevin Romero
 * Purpose: Implementation of Player Object for Blackjack
 * Created on July 12, 2017, 5:16 PM
 */

//System Libraries

//User Libraries
#include "Player.h"

//Implement Default Constructor
Player::Player(){
    this->isBust();
    this->firstDeal();
}

//Copy deck
Player::Player(Shoe *s){
    this->aShoe = s;
    this->isBust();
    this->firstDeal();
}

//default win
bool Player::winBJ(){
    return (this->aHand.getHandTotal()==21 && this->aHand.getNumCards() == 2);
}

//Det. if player bust
bool Player::isBust(){
    if(aHand.getHandTotal()>21)
        return true;
    return false;
}

//Get total hand score
const int Player::getTotScore(){
    return aHand.getHandTotal();
}

//Print cards in hand
void Player::prntHand(){
    aHand.printHand();
}

//Implement Constructor
void Player::firstDeal(){
    aHand.initDraw(aShoe->dealCard(),aShoe->dealCard());
}

//take another card
void Player::hit(){
    aHand.addCard(aShoe->dealCard());
}

//Print the deck
void Player::viewShoe(){
    aShoe->printShoe();
}

File: GrpProj1/BlackjackJS/Player.js
/*
	File: Player.js 
	Author: Kevin Romero
	Class: CSC17B Team Blackjack
	Purpose: Player object for Blackjack
	Date: 10/16/17
*/

//create player class
function Player(){
	//member variables
	this.aShoe = new Shoe();
	this.aHand = new Hand();
	this.bust;

	//this.viewShoe();
	//this.firstDeal();
	//this.prntHand();
	//document.write("Hand Score: "+this.getTotScore()+"<br>");
	//document.write("Card in Hand: "+this.getCardsInHand()+"<br>");

	//constructor functions

};

Player.prototype.constructor=function(shoe){
	this.aShoe = shoe;
	this.isBust();
};

Player.prototype.winBJ=function(){
	if(this.getTotScore()==21 && this.getCardsInHand()==2){
		return true;
	}
};

Player.prototype.isBust=function(){
	if(this.getTotScore()>21){
		this.bust=true;
	}else{
		this.bust=false;
	}
};

Player.prototype.firstDeal=function(){
	this.aHand.initDraw(this.aShoe.dealCard(),this.aShoe.dealCard());
};

Player.prototype.getTotScore=function(){
	return this.aHand.getHandTotal();
};

Player.prototype.prntHand=function(){
	this.aHand.printHand();
};

Player.prototype.getCardsInHand=function(){
	return this.aHand.getNumCards();
};

//take another card;
Player.prototype.hit=function(){
	this.aHand.addCard(this.aShoe.dealCard());
};

//print deck
Player.prototype.viewShoe=function(){
	this.aShoe.printShoe();
};


